### DISCUSSION TITLE
Allow linking text in code blocks

### DISCUSSION THREAD

**Discussion by Babilinski** [facebook/docusaurus#6152] [Ideas]
I'm creating a documentation page which includes API references. I wish Docusaurus provided a guide or feature that would allow me to add links to code blocks so user's could get more information about a type listed in the code by clicking on it.

**Comment by Josh-Cena** [comment]
Docusaurus site itself would also benefit from this feature, but it isn't standard Markdown convention. Code blocks are shipped out as `pre` elements and its contents are untouched. You may need to [swizzle `CodeBlock`](https://docusaurus.io/docs/sing-themes#swizzling-theme-components) or [make a remark plugin](https://docusaurus.io/docs/markdown-features/plugins#creating-new-rehyperemark-plugins) to implement what you have in mind.

**Reply by Babilinski** [reply]
I would prefer if they were untouched, when I try to use the pre tags, the end result looks like this:

```
[object Object][object Object][object Object]
[object Object]
[object Object] [object Object] GetLongestDistance [object Object]
  first_distance [object Object] [object Object][object Object]
   [object Object]
[object Object][object Object][object Object]
```

**Reply by Josh-Cena** [reply]
Yes, see #6156.

**Reply by Josh-Cena** [reply]
The `[object Object]` problem is fixed by #6177. Please [test with the latest canary release](https://docusaurus.io/community/canary).
